qemu/tests/block_iothread_test.py

```diff
@@ -0,0 +1,152 @@
+"""Block device iothread relevant test"""
+import re
+
+from provider.block_devices_plug import BlockDevicesPlug
+from virttest import error_context
+from virttest import env_process, utils_disk, utils_misc, virt_vm
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+from aexpect import ShellCmdError
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Block device iothread relevant test
+    The test include boot positive and negative
+    Positive:
+    1) Boot Vm data disk enable iothread.
+    2) Check disk iothread set.
+    3) unplug and hotplug disk.
+    4) Simple IO on disk.
+    5) Check disk iothread set.
+    Negative:
+    1) Boot Vm data disk with wrong configration.
+    2) Catch exception to check the message whether is expected ?
+    """
+
+    def _get_window_disk_index_by_serial(serial):
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+        test.fail("Not find expected disk %s" % serial)
+
+    def _check_disk_in_guest(img):
+        os_type = params["os_type"]
+        logger.debug("Check disk %s in guest" % img)
+        if os_type == 'windows':
+            img_size = params.get("image_size_%s" % img)
+            cmd = utils_misc.set_winutils_letter(session, guest_cmd)
+            disk = _get_window_disk_index_by_serial(img)
+            utils_disk.update_windows_disk_attributes(session, disk)
+            logger.info("Clean disk:%s", disk)
+            utils_disk.clean_partition_windows(session, disk)
+            logger.info("Formatting disk:%s", disk)
+            driver = \
+                utils_disk.configure_empty_disk(session, disk, img_size,
+                                                os_type)[0]
+            output_path = driver + ":\\test.dat"
+            cmd = cmd.format(output_path)
+        else:
+            output_path = get_linux_drive_path(session, img)
+            cmd = guest_cmd.format(output_path)
+
+        logger.debug(cmd)
+        session.cmd(cmd)
+
+    def check_image_iothread():
+        images = params.get("check_iothread_images", "").strip().split()
```

Comment:
The following method can also work.
images = params.get_list("check_iothread_images")